scripts/content-utils/card-tree.ts

```diff
@@ -21,72 +21,135 @@ export type CardLeaf = {
     modifiers?: GameWorldModifier | GameWorldModifier[]
 }
 
+// TODO: Refactor CardTree to use this runtime type guard
+export function isCardTree(node: any): node is CardTree {
+    return node && node.card !== undefined
+}
+
+/**
+ * @param tree The CardTree to recursively turn into an array of cards
+ *
+ * NOTE: The following params should only be used internally by the content utils
```

Comment:
I would like to avoid this `NOTE` by separating this into two different functions.

Reply:
This would be a good next PR I think.